test/map_notifier_test.dart

```diff
@@ -131,41 +163,60 @@ void main() {
 
       test('Listener is not notified if no value is removed', () {
         mapNotifier.addAll(newValues);
+
+        /// first call when addAll is called
         expect(listenerCallCount, 1);
         mapNotifier.remove('zero');
+
+        /// second call when remove is called
         expect(listenerCallCount, 2);
         mapNotifier.remove('zero');
+
+        /// The count is still 2 because the value is already removed
         expect(listenerCallCount, 2);
       });
 
       test(
         'Listener is not notified if removeWhere does not remove any values',
         () {
           mapNotifier.addAll(newValues);
+
+          /// first call when addAll is called
           expect(listenerCallCount, 1);
           mapNotifier.removeWhere((key, _) => key == 'ten');
+
+          /// The count is still 1 because the value is not excited in mapNotifier
           expect(listenerCallCount, 1);
         },
       );
 
       test(
           'Listener is not notified when update is called and updates to the '
           'already existing value', () {
+        /// The key exist in mapNotifier, so the value is not added
         mapNotifier.update('zero', (_) => 10, ifAbsent: () => 10);
-        expect(listenerCallCount, 1);
+
+        /// The listener isn't called because the value is already updated
+        expect(listenerCallCount, 0);
         mapNotifier.update('zero', (_) => 10, ifAbsent: () => 10);
-        expect(listenerCallCount, 1);
```

Comment:
as we don't reset the listenerCallCount after the last all it should still be 1 here. but I agree it would probably be more expressive if we rest it to 0 before the call to `update`